quarto/01-QC-DMT01.qmd

```diff
@@ -0,0 +1,161 @@
+
+```{r, warning=FALSE}
+rtables_result <- rtables::as_result_df(tlg_dmt01, make_ard = TRUE)
+rtables_result[1:6, -c(2:4)]
+```
+
+#### 3. Create a comparable ARD
+
+Using the {cards} package, we stack the functions `ard_continuous()` for the continuous variables and `ard_categorical()` for categorical variables. The default statistics calculated for each of these data types are included - these can be adapted for bespoke analyses.
+
+If any variable manipulation was done to the data prior to running the citril/chevron/tern/rtables commands, we suggest supplying the same data to these ARD functions, or running the same pre-processing steps to your data prior to creating ARDs to ensure variable names/levels match.
+
+Note: when a 'by' variable is specified in the  `ard_stack()` call, a univariate tabluation is returned. We will have to remove those obervations later in the workflow to create a 1:1 match. 
+
+```{r}
+#| code-fold: false
+#| code-summary: "Show the code"
+library(cards)
+
+# build ARDs that calculate relevant statistics for continuous and categorical variables.
+
+ard_result <- 
+  ard_stack(
+    syn_data$adsl, 
+    ard_continuous(variables = c(AGE), 
+                   statistic = ~ continuous_summary_fns(c("N", "mean", "sd", "median", "min", "max"))),
+    ard_categorical(variables = c(AGEGR1, SEX, ETHNIC, RACE), statistic =everything() ~ c("n", "p")),
+    ard_missing(variables = c(AGEGR1, SEX, ETHNIC, RACE), statistic = everything()~ c("N_obs")),
+    .by = "ARM",
+    .overall = TRUE)
+
+ard_result [1:6, -c(1,9:11)]
+```
+
+### Statistics comparison
+
+#### 1. rtables reformat
+
+In order to compare the two data.frames programatically, some identifying variables must align to be used as "key columns". Below are some data wrangling steps used to match the statistics for comparison. Note the rtables output:
+
+```{r}
+tail(rtables_results[, -c(1:3)])
+```
+
+The `variable_level` leads with the statistic name, followed by "." and then the actual level that matches the `variable_level` in the ARD object. We will mutate the level to match the level values in the ard object. Similarly, the total number of observations for a group is labelled "N" in the ARD object, while it is named "n" in the rtables object. The following manipulations are completed below:
+
+1.  Set the `variable_level` to NA in the rtables result for a continuous data summary (as variable levels don't apply and is NULL in the ARD object).
+2.  Remove the stat_name (ie. "count") before the variable level.
+3.  Convert the "n":"N" and "count":"n".
+4.  Recode the variable AAGE to AGE.
+5.  Recode the `SEX` variable Male:M, Female:F. Note: if you are mapping variables in {citril}, these changes would already be captured.
+6.  Let's remove columns we know won't be in the ard dataframe for simplicity (ie. stat_string and variable label)
+
+```{r}
+
+rtables_result <- rtables_result |>
+  dplyr::mutate(
+    variable = dplyr::recode(variable, "AAGE" = "AGE"),
```

Comment:
If you use the same data this would be the same

Reply:
I think it is important to use the same data. If you check the chevron function there is AAGE and not AGE. It is the same var